src/domains/Lotto.js

```diff
@@ -0,0 +1,132 @@
+const MissionUtils = require('@woowacourse/mission-utils');
+const Messages = require('../constants/Messages');
+const LottoError = require('../errors/LottoError');
+
+class Lotto {
+  /** 로또가 가질 수 있는 숫자 갯수 */
+  static NUMBER_COUNT = 6;
+
+  /** 로또가 가질 수 있는 숫자의 최소값 */
+  static NUMBER_MIN = 1;
+
+  /** 로또가 가질 수 있는 숫자의 최대값 */
+  static NUMBER_MAX = 45;
+
+  /** 로또 1장의 가격 */
+  static PRICE = 1000;
+
+  /** @type {number[]} */
+  #numbers;
+
+  /**
+   * @param {number[]} numbers
+   */
+  constructor(numbers) {
+    numbers.sort((a, b) => a - b);
+    this.#numbers = numbers;
+    this.validate();
+  }
+
+  /**
+   * 무작위 번호를 가지는 로또를 생성한다.
+   *
+   * @returns {Lotto}
+   */
+  static fromRandom() {
+    const numbers = MissionUtils.Random.pickUniqueNumbersInRange(
+      Lotto.NUMBER_MIN,
+      Lotto.NUMBER_MAX,
+      Lotto.NUMBER_COUNT,
+    );
+    return new Lotto(numbers);
+  }
+
+  /**
+   * 문자열로부터 숫자를 파싱하여 로또를 생성한다.
+   *
+   * @param {string} text
+   * @returns {Lotto}
+   */
+  static fromString(text) {
+    return new Lotto(text.split(',').map(Number));
+  }
+
+  /**
+   * @param {number} money
+   * @returns {Lotto[]}
+   */
+  static buyLottos(money) {
+    if (money < Lotto.PRICE) {
+      throw new LottoError(Messages.LOTTO_BUY_AT_LEAST_ONE);
+    }
+    if (money % Lotto.PRICE !== 0) {
+      throw new LottoError(Messages.LOTTO_BUY_NO_CHANGE);
+    }
+
+    const amount = money / Lotto.PRICE;
+    return Array(amount)
+      .fill()
+      .map(() => Lotto.fromRandom());
+  }
+
+  /**
+   * @param {number[]} numbers
+   */
+  validate() {
+    this.#validateType();
+    this.#validateNumberCount();
+    this.#validateNumberRange();
+    this.#validateDuplication();
+  }
+
+  /**
+   * @param {number} number
+   * @returns {boolean}
+   */
+  hasNumber(number) {
+    return this.#numbers.includes(number);
+  }
+
+  /**
+   * @returns {number[]}
+   */
+  getNumbers() {
+    return this.#numbers;
+  }
+
+  toString() {
+    return `[${this.#numbers.join(', ')}]`;
+  }
+
+  #validateType() {
+    if (!this.#numbers.every((number) => typeof number === 'number' && !Number.isNaN(number))) {
```

Comment:
if 문들을 따로 캡슐화해서 분리했으면 좋았을 것 같아요!

Reply:
[조건 캡슐화](https://github.com/labs42io/clean-code-typescript#encapsulate-conditionals) 글 읽어보시면 좋을 것 같아요 !

Reply:
아하! 감사합니다. 저걸 별도 함수로 만들면 되겠군요

Reply:
@Leemainsw 님께서 좋은 지적해주신 것 같아요! 저도 지나가다 공유하고 싶은 의견이 있는데요!

저는 #validateType이 충분히 조건 캡슐화가 되었다 생각해요. (네이밍은 #validateNumberType이 더 좋겠지만요 ㅎㅎ)
왜냐면 첨부해주신 글이나 [자바스크립트 조건 캡슐화](https://github.com/ryanmcdermott/clean-code-javascript#encapsulate-conditionals) 글까지 확인을 해보았을 때 각각 다른 파라미터를 비교해 다른 사람이 보았을 때 파악하기 어려운 조건식이기에 조건식만 분리하는 것처럼 느껴졌거든요.

이번에 용혁님께서 작성해주신 코드에선 **Number 타입인지 유효성 검사** 하는 조건식이기에 만약 분리한다면 아래와 같이 하는 건 어떨까 생각이 드네요!
```suggestion
    const isNumberArray = this.#numbers.every((number) => typeof number === 'number';
    if (!isNumberArray && !Number.isNaN(number))) {
```

Reply:
리팩토링을 위해 조건 캡슐화를 하려니 거의 동일한 함수를 만들어야겠더라구요. 캡슐화가 이미 충분히 되어있다는 것이겠죠?

Reply:
맞습니다!!!! 저는 충분히 되어있다 생각이 들어요 ;)

Reply:
감사합니다!! 잘 설명해주신 덕분에 어떠한 경우에 캡슐화를 하는게 좋을 지 알게 된 것 같아요. 👍👍